packages/components/src/button/index.tsx

```diff
@@ -65,10 +65,9 @@ function useDeprecatedProps( {
 	}
 
 	if ( isDefault ) {
-		deprecated( 'Button isDefault prop', {
+		deprecated( 'wp.components.Button `isDefault` prop', {
 			since: '5.4',
 			alternative: 'variant="secondary"',
-			version: '6.2',
```

Comment:
Should we at least bump this so we still remove it in some future WP version?

Reply:
It's the ambiguous back compat policy biting us again 😅

Up to this point the components package hasn't done many hard deprecations, mostly because a lot of these back compat layers are very low cost. Like virtually zero cost compared to the migration management cost we'd push to our third-party consumers who have to support multiple WP versions, and also the cost incurred on the end users who see incorrectly styled UI from all the consumers that never migrate.

And in that vein, I'm not particularly enthusiastic about logging deprecation warnings on "simple translation" API changes like this that really add no value to anyone. _If_ we're not planning to hard deprecate, then there's no reason for consumers to migrate to the new API unless they want to for ergonomics or code quality reasons. I've been hearing that even managing these deprecation warnings while supporting multiple WP versions is cumbersome, so I don't consider them "free" even though they're dev-env only.

Happy to discuss and explicitly codify things a bit more, but I'm in no rush to hard deprecate things unless they have a tangible maintenance cost.